utilities/general-helpers.metta

```diff
@@ -1142,3 +1141,95 @@
                   $then
             )
             ))
+
+;; It returns True only if the input is zero.
+(: isZero (-> Number Bool))
+(= (isZero $x) (== $x 0))
+
+;; It will return True if greater than 10.
+(: greaterThanTen (-> Number Bool))
+(= (greaterThanTen $x) (> $x 10))
+
+;; It will decrement the input by 1.
+(: decrement (-> Number Number))
+(= (decrement $x) (- $x 1))
+
+;; It will increment the input by 1.
+(: increment (-> Number Number))
+(= (increment $x) (+ $x 1))
+
+;; A function that turns every atom to unit
+(: transformToUnit (-> Atom Atom))
+(= (transformToUnit $x) ())
+
+; %%%% Hyperon-Experimental compatibility library. %%%%
+; Many of the included functions were considered ad hoc and
+; straightforward to express directly in MeTTa itself:
+
+;Equality and Reduction
+(= (if-equal $A $B $Then $Else)
+   (if (== $A $B) $Then $Else))
+
+(= (if-equal2 $A $B $Then $Else)
+   (if (== $A $B) $Then $Else))
+
+(= (assertEqual $A $B)
+   (assert (== $A $B)))
+
+(= (assertAlphaEqual $A $B)
+   (assert (=alpha $A $B)))
+
+(: unify (-> Expression Expression Expression Expression %Undefined%))
+(= (unify $a $b $then $else)
+   (if (= $a $b)
+       (eval $then)
+       (eval $else)))
+
+(: add-reduct (-> %Undefined% Expression Bool))
+(= (add-reduct $space $f)
+   (let* (($headbody (cdr-atom $f))
+          ($head (car-atom $headbody))
+          ($body (cdr-atom $headbody))
+          ($bodyreduced (eval $body)))
+         (add-atom $space (= $head $bodyreduced))))
```

Comment:
I don't think this `add-reduct` has the same behavior as the one we have been using. Was that intentional?